src/unstable/components/Form/index.tsx

```diff
@@ -74,44 +74,43 @@ const FormWrapper = styled(Box)`
 	}
 `;
 
-export default class FormHOC extends React.Component<
-	FormProps,
-	{ value: any; schema: JSONSchema6 }
-> {
+const parseSchema = (schema: JSONSchema6) => {
+	const whitelist = SUPPORTED_SCHEMA_FORMATS.concat(Object.keys(widgets));
+	return utils.stripSchemaFormats(schema, whitelist);
+};
+
+interface FormState {
+	value: any;
+	schema: JSONSchema6;
+}
+
+export default class FormHOC extends React.Component<FormProps, FormState> {
 	private formRef = React.createRef<Form<any>>();
 
-	constructor(props: FormProps) {
-		super(props);
+	static getDerivedStateFromProps(props: FormProps, state: FormState) {
+		if (!state) {
+			return { schema: parseSchema(props.schema), value: props.value };
+		}
 
-		this.state = {
-			value: props.value || (props.schema.type === 'array' ? [] : {}),
-			schema: this.parseSchema(props.schema),
-		};
+		if (!isEqual(state.value, props.value)) {
+			return {
```

Comment:
I would actually prefer for this to be a only controlled component without a local state, but this will do for now.